notes/design_principles.md

```diff
@@ -0,0 +1,101 @@
+
+## Design Principles for Data Modeling in HyBox
+
+### Use discovery and search requirements as a pragmatic means to determine the needed extent of the modeling efforts
```

Comment:
Do we include workflow requirements ("librarian" use cases), here? 

Equivocation on this point over the years has bitten FR* modeling efforts over the years, and shows every sign of doing so again with the current `FRBR LRM` release. This would be a good trap to avoid.

e.g. assuming a (spurious) use case where staff need to retrieve items for metadata review based on size, does that motivate a preference for the model proceeding from https://github.com/hybox/models/pull/20/files#diff-bbc93c64e140c58aee05a39058e8901dR14 over https://github.com/hybox/models/pull/20/files#diff-bbc93c64e140c58aee05a39058e8901dR8 ?